README.md

```diff
@@ -1 +1,101 @@
-# miden-node
\ No newline at end of file
+# Miden node
+
+<a href="https://github.com/0xPolygonMiden/miden-node/blob/main/LICENSE"><img src="https://img.shields.io/badge/license-MIT-blue.svg"></a>
+<img src="https://github.com/0xPolygonMiden/miden-node/workflows/CI/badge.svg?branch=main">
+<a href="https://crates.io/crates/miden-node"><img src="https://img.shields.io/crates/v/miden-node"></a>
+
+This repository holds the Miden node; that is, the software which processes transactions and creates blocks for the Miden rollup.
+
+### Status
+
+The Miden node is still under heavy development and the project can be considered to be in an *alpha* stage. Many features are yet to be implemented and there is a number of limitations which we will lift in the near future.
+
+At this point, we are developing the Miden node for a centralized operator. Thus, the work does not yet include such components as P2P networking and consensus. These also will be added in the future.
+
+## Architecture
+
+![Architecture diagram](./assets/architecture.png)
+
+The Miden node is made up of 3 main components, which communicate over gRPC: 
+- **store:** stores the current state of the chain.
+- **rpc:** serves client requests such as to synchronize with the latest state of the chain or to submit transactions.
+- **block producer:** accepts transactions from the RPC component, creates blocks containing those transactions, and sends them to the store.
+
+All 3 components can either run in one process, or each component can run in its own process. See the [Running the node](#running-the-node) section for more details.
+
+## Usage
+
+Before running the node, you must first generate the genesis file. 
+
+The `miden-node` executable is used to both generate the genesis file, and running the node.
+
+### Installing the node
+
+To install for production use cases, run
+
+```sh
+# Installs `miden-node` executable
```

Comment:
Do we need this comment here? Seems like it duplicates content on line 34 above.

I would also mention what the outcome of this command (i.e., you can run miden node from anywhere - or something better worded than this) is as not everyone will be familiar with what `cargo install` does.

Reply:
done